Makefile

```diff
@@ -0,0 +1,52 @@
+# Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+
+SHELL=/bin/bash -o pipefail
+
+PROJECT_NAME=zookeeper-operator
+REPO=pravega/$(PROJECT_NAME)
+VERSION=$(shell git describe --always --tags --dirty)
+GIT_SHA=$(shell git rev-parse --short HEAD)
+GOOS=linux
+GOARCH=amd64
+
+.PHONY: all build check clean test
+
+all: check build
+
+build: test build-go build-image
+
+build-go:
+	CGO_ENABLED=0 GOOS=$(GOOS) GOARCH=$(GOARCH) go build \
+	-ldflags "-X github.com/$(REPO)/pkg/version.Version=$(VERSION) -X github.com/$(REPO)/pkg/version.GitSHA=$(GITSHA)" \
+	-o bin/$(PROJECT_NAME) cmd/$(PROJECT_NAME)/main.go
+
+build-image:
+	docker build --build-arg VERSION=$(VERSION) --build-arg GIT_SHA=$(GIT_SHA) -t $(REPO):$(VERSION) .
+	docker tag $(REPO):$(VERSION) $(REPO):latest
+
+test:
+	go test $$(go list ./... | grep -v /vendor/)
+
+login:
```

Comment:
Who provides these env vars during travis build?

Reply:
These Docker variables are configured in the Travis job, so they are present during the build process.